### DISCUSSION TITLE
Wire model with debounce lose my input focus when the component is reloaded

### DISCUSSION THREAD

**Discussion by pirmax** [livewire/livewire#5013] [1. Help]
Hi,

My wire model (with debounce) lose my search input focus when the component is reloaded (search done).

`<input wire:model.debounce.1000ms="search" type="search" name="search" id="search" class="">`

I use query string to save the query in URL (for the back button for example).

The exact problem is that when I enter a value in the search field, the component is re-rendered but the focus of the input is lost, sometimes after a second.

```
class WelcomeComponent extends Component
{
    use WithPagination;

    public string $search = '';

    protected $queryString = [
        'search' => [
            'except' => '',
        ],
    ];

    public function render()
    {
        $items = Item::search($this->search)->paginate(15);

        return view('livewire.app.welcome-component', [
            'items' => $items,
        ])->layout('layouts.app');
    }
}
```

The search field is just above the list (array of values), and therefore in the same component, can it come from there (this problem)?

I'd like to not lose focus to continue searching if the user pauses while typing.

Thanks.

**Comment by joshhanley** [comment]
@pirmax yeah it shouldn't be doing that! Any chance you can try and reproduce the issue on https://laravelplayground.com/ so we can have a look?

Thanks!
